### DISCUSSION TITLE
No pending certificate

### DISCUSSION THREAD

**Discussion by adavault-ops** [input-output-hk/mithril#496] [Q&A] [answered]
Mithril seems to be set up correctly(?), but throws "no pending certificate" error. Node is synced and running on preview network.
Any ideas what I did wrong?

● mithril-signer.service - Mithril Signer service
     Loaded: loaded (/etc/systemd/system/mithril-signer.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2022-09-07 17:09:58 UTC; 3 days ago
   Main PID: 223992 (mithril-signer)
      Tasks: 14 (limit: 19104)
     Memory: 206.7M
     CGroup: /system.slice/mithril-signer.service
             └─223992 /opt/cardano/mithril/mithril-signer -vvv

Sep 11 10:37:58 vducdn53 mithril-signer[223992]: {"msg":"Retrieve pending certificate","v":0,"name":"slog-rs","level":20,"time":"2022-09-11T10:37:58.748765277Z","hostname":"vducdn53","pid":223992}
Sep 11 10:37:58 vducdn53 mithril-signer[223992]: {"msg":" ⋅ no pending certificate, waiting…","v":0,"name":"slog-rs","level":20,"time":"2022-09-11T10:37:58.875235183Z","hostname":"vducdn53","pid":223992}
Sep 11 10:37:58 vducdn53 mithril-signer[223992]: {"msg":"Sleeping for 60000 ms","v":0,"name":"slog-rs","level":30,"time":"2022-09-11T10:37:58.875271423Z","hostname":"vducdn53","pid":223992}
Sep 11 10:38:58 vducdn53 mithril-signer[223992]: {"msg":"================================================================================","v":0,"name":"slog-rs","level":30,"time":"2022-09-11T10:38:58.87581603Z","hostname":"vducdn53","pid":223992}
Sep 11 10:38:58 vducdn53 mithril-signer[223992]: {"msg":"STATE MACHINE: new cycle","v":0,"name":"slog-rs","level":20,"time":"2022-09-11T10:38:58.875862785Z","hostname":"vducdn53","pid":223992,"current_state":"Registered(RegisteredState { beacon: Beacon { network: \"preview\", epoch: Epoch(33), immutable_file_number: 656 } })"}
Sep 11 10:38:58 vducdn53 mithril-signer[223992]: {"msg":"RUNNER: get_current_epoch","v":0,"name":"slog-rs","level":20,"time":"2022-09-11T10:38:58.875875472Z","hostname":"vducdn53","pid":223992}
Sep 11 10:38:58 vducdn53 mithril-signer[223992]: {"msg":"RUNNER: get_pending_certificate","v":0,"name":"slog-rs","level":20,"time":"2022-09-11T10:38:58.892415911Z","hostname":"vducdn53","pid":223992}
Sep 11 10:38:58 vducdn53 mithril-signer[223992]: {"msg":"Retrieve pending certificate","v":0,"name":"slog-rs","level":20,"time":"2022-09-11T10:38:58.892634455Z","hostname":"vducdn53","pid":223992}
Sep 11 10:38:59 vducdn53 mithril-signer[223992]: {"msg":" ⋅ no pending certificate, waiting…","v":0,"name":"slog-rs","level":20,"time":"2022-09-11T10:38:59.017437057Z","hostname":"vducdn53","pid":223992}
Sep 11 10:38:59 vducdn53 mithril-signer[223992]: {"msg":"Sleeping for 60000 ms","v":0,"name":"slog-rs","level":30,"time":"2022-09-11T10:38:59.01772316Z","hostname":"vducdn53","pid":223992}

**Comment by jpraynaud** [comment, marked as answer]
Hi @adavault, this is actually the expected behavior:
* A `Pending Certificate` is created each time a new `Immutable File` is appended to the database folder of the Cardano node (which happens every `~2` hours on the `preview` network)
* The `Pending Certificate` is available until a new `Certificate` is created (which happens when the `Mithril Aggregator` has received enough `Individual Signatures` from the `Mithril Signers`)

As soon as you see that your `Signer` node is in the `Registered` state (`Registered(RegisteredState { beacon: Beacon { network: "preview", epoch: Epoch(33), immutable_file_number: 656 } })` in your logs), it means that your node will be able to contribute to the creation of the multi-signatures.

**Reply by adavault-ops** [reply]
Thanks for taking the time to confirm.
